Bloat regression CI gate broken since shipping (setup-uv@v3 cache error)

## Goal

Restore the bloat regression CI gate (Stage 6 of #2886) so it actually runs `tests/test_esp32s3_bloat_regression.py` instead of failing in setup.

Parent: #2886 — measurement infrastructure.

## Symptom

Every recent run of \`bloat regression esp32s3\` workflow fails in ~10s at the **Set up uv (Python toolchain)** step before checkout's uv.lock can be matched:

\`\`\`
##[error]No file in /home/runner/work/FastLED/FastLED matched to [**/uv.lock], make sure you have checked out the target repository
\`\`\`

The 8 most recent runs (on PRs and master pushes alike) all fail this way:

| Run | Branch | Event | Conclusion |
|---|---|---|---|
| 27093491519 | feat/2942-gate-channel-makename | PR | fail |
| 27089004680 | master | push | fail |
| 27089003970 | feat/2931-disable-channel-events | PR | fail |
| 27088237650 | master | push | fail |
| 27088236849 | chore/ratchet-baseline-339962 | PR | fail |
| 27088193057 | master | push | fail |
| 27088192155 | feat/2928-gamma-single-entry-cache | PR | fail |
| 27087438075 | master | push | fail |

This means the verification gate that #2886 Stage 6 was supposed to enforce has been silently no-op since it shipped — bloat-regressing PRs would not be flagged.

## Root cause

\`.github/workflows/bloat_regression_esp32s3.yml:69\` pins \`astral-sh/setup-uv@v3\`. Every other workflow in the repo that uses setup-uv pins \`@v6\`:

\`\`\`
$ grep -r 'astral-sh/setup-uv' .github/workflows
.github/workflows/avr8js_docker_template.yml:        uses: astral-sh/setup-uv@v6
.github/workflows/bloat_regression_esp32s3.yml:     uses: astral-sh/setup-uv@v3   <-- outlier
.github/workflows/build_template.yml:               uses: astral-sh/setup-uv@v6
.github/workflows/header-perf.yml:                  uses: astral-sh/setup-uv@v4
.github/workflows/iwyu.yml:                         uses: astral-sh/setup-uv@v6
.github/workflows/qemu_docker_template.yml:         uses: astral-sh/setup-uv@v6
.github/workflows/template_example_test.yml:       uses: astral-sh/setup-uv@v6
.github/workflows/template_unit_test.yml:          uses: astral-sh/setup-uv@v6
\`\`\`

setup-uv@v3's cache-key glob (\`**/uv.lock\`) errors when the glob returns zero matches even though \`uv.lock\` is present in the repo at HEAD (verified locally). This is a known v3 quirk that was fixed in v4+.

## Deliverable

### \`.github/workflows/bloat_regression_esp32s3.yml\` — bump setup-uv

\`\`\`diff
 - name: Set up uv (Python toolchain)
-  uses: astral-sh/setup-uv@v3
+  uses: astral-sh/setup-uv@v6
   with:
     enable-cache: true
\`\`\`

## Acceptance

1. \`bloat regression esp32s3\` workflow run on this fix's PR reaches the **Run ESP32-S3 bloat regression gate** step and either:
   - Passes (flash ≤ pinned baseline), or
   - Fails legitimately with a flash regression message from \`tests/test_esp32s3_bloat_regression.py\` — not from setup-uv.
2. Master push runs after merge stop failing in setup.

## Out of scope

- Reviewing whether the pinned baseline needs ratcheting after the recent stream of PRs (#2918/#2925/#2929/#2942). Once the gate is live again the next push run will surface any drift; ratcheting belongs in a follow-up.
- Pinning every setup-uv usage in the repo to a single version (the others are working fine).

## Related

- **#2886** — parent meta (Stage 6 verification gate).
- **#2900** — original Stage 6 PR that shipped this workflow.
- **#2942 / #2943** — the most recent PR where this CI failure surfaced.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/.github/workflows/bloat_regression_esp32s3.yml
+++ b/.github/workflows/bloat_regression_esp32s3.yml
@@ -66,7 +66,7 @@ jobs:
       - uses: actions/checkout@v4
 
       - name: Set up uv (Python toolchain)
-        uses: astral-sh/setup-uv@v3
+        uses: astral-sh/setup-uv@v6
         with:
           enable-cache: true
 

